packages/twenty-front/src/modules/object-record/record-index/hooks/useLoadRecordIndexStates.ts

```diff
@@ -0,0 +1,281 @@
+              );
+              return {
+                ...column,
+                isFilterable: existsInFilterDefinitions,
+                isSortable: existsInSortDefinitions,
+              };
+            });
+
+        const newFieldDefinitions = mapViewFieldsToColumnDefinitions({
+          viewFields,
+          columnDefinitions,
+        });
+
+        setTableColumns(newFieldDefinitions, recordIndexId);
+
+        const existingRecordIndexFieldDefinitions = snapshot
+          .getLoadable(recordIndexFieldDefinitionsState)
+          .getValue();
+        if (
+          !isDeeplyEqual(
+            existingRecordIndexFieldDefinitions,
+            newFieldDefinitions,
+          )
+        ) {
+          set(recordIndexFieldDefinitionsState, newFieldDefinitions);
+        }
+
+        for (const viewField of viewFields) {
+          const viewFieldMetadataType = objectMetadataItem.fields?.find(
+            (field) => field.id === viewField.fieldMetadataId,
+          )?.type;
+
+          const aggregateOperationForViewField = snapshot
+            .getLoadable(
+              viewFieldAggregateOperationState({
+                viewFieldId: viewField.id,
+              }),
+            )
+            .getValue();
+          const convertedViewFieldAggregateOperation = isDefined(
+            viewField.aggregateOperation,
+          )
+            ? convertAggregateOperationToExtendedAggregateOperation(
+                viewField.aggregateOperation,
+                viewFieldMetadataType,
+              )
+            : viewField.aggregateOperation;
+
+          if (
+            aggregateOperationForViewField !==
+            convertedViewFieldAggregateOperation
+          ) {
+            set(
+              viewFieldAggregateOperationState({
+                viewFieldId: viewField.id,
+              }),
+              convertedViewFieldAggregateOperation,
+            );
+          }
+        }
+      },
+    [setTableColumns],
+  );
+
+  const onViewGroupsChange = useCallback(
+    (
+      viewGroups: ViewGroup[],
+      objectMetadataItem: ObjectMetadataItem,
+      recordIndexId: string,
+    ) => {
+      const newGroupDefinitions = mapViewGroupsToRecordGroupDefinitions({
+        objectMetadataItem,
+        viewGroups,
+      });
+
+      setRecordGroup(newGroupDefinitions, recordIndexId);
+    },
+    [setRecordGroup],
+  );
+
+  const loadRecordIndexStates = useRecoilCallback(
+    ({ snapshot, set }) =>
+      async (view: View, objectMetadataItem: ObjectMetadataItem) => {
+        const recordIndexId = `${objectMetadataItem.namePlural}-${view.id}`;
+
+        const filterableFieldMetadataItems = snapshot
+          .getLoadable(
+            availableFieldMetadataItemsForFilterFamilySelector({
+              objectMetadataItemId: objectMetadataItem.id,
+            }),
+          )
+          .getValue();
+
+        onViewFieldsChange(view.viewFields, objectMetadataItem, recordIndexId);
```

Comment:
That hook is doing too much and those handlers shouldn't be called like that, seems like a complexity zone that could be refactored, if it works for now let's keep it like that but let's think about how we could break this down.